core/src/memory.rs

```diff
@@ -44,6 +45,28 @@ pub struct MemoryInterval {
     pub size: u32,
 }
 
+impl MemoryInterval {
+    /// Convert `MemoryInterval` to `[u8; 8]` bytes.
+    /// `0..4` - `offset`
+    /// `4..8` - `size`
+    pub fn to_bytes(&self) -> [u8; 8] {
+        let mut bytes = [0u8; 8];
+        LittleEndian::write_u32(&mut bytes[0..4], self.offset);
+        LittleEndian::write_u32(&mut bytes[4..8], self.size);
+        bytes
+    }
```

Comment:
Could we make this function `const` using `to_le_bytes` and forming the resulting array from bytes?

Reply:
https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=9f6c939fe9159c1bc42359f9ac5a47ef

Reply:
idk why. But approach with const fn is slower than `LittleEndian`. Same for `from_bytes`